modules/sc-mesh-secure-deployment/src/nats/tests/test_cbma_adaptation.py

```diff
@@ -0,0 +1,505 @@
+        self.cbma_adaptation._CBMAAdaptation__add_interface_to_bridge(
+            "br-lan", "vlan_red"
+        )
+        # Success means no debug traces in tested function
+        self.cbma_adaptation.logger.debug.assert_not_called()
+        self.cbma_adaptation.logger.error.assert_not_called()
+
+    def test_set_interface_up(self):
+        # vlan_red_ create in cbma_adaptation instantiation,
+        # setting it up should pass
+        self.cbma_adaptation._CBMAAdaptation__set_interface_up("vlan_red")
+        self.cbma_adaptation.logger.error.assert_not_called()
+
+        # Setting up not existing interface should fail
+        self.cbma_adaptation._CBMAAdaptation__set_interface_up("vlan_blue")
+        self.cbma_adaptation.logger.error.assert_called_once()
+
+    def test_get_mac_addr(self):
+        self.patcher_get_mac_addr.stop()
+        # Update interfaces list from where to get MAC
+        self.cbma_adaptation._CBMAAdaptation__get_interfaces()
+        # Get MAC
+        result = self.cbma_adaptation._CBMAAdaptation__get_mac_addr("vlan_red")
+        self.assertIsNotNone(result)
+        result = self.cbma_adaptation._CBMAAdaptation__get_mac_addr("foobar")
+        self.assertIsNone(result)
+        self.patcher_get_mac_addr.start()
+
+    def test_has_certificate(self):
+        self.patcher_has_certificate.stop()
+        result = self.cbma_adaptation._CBMAAdaptation__has_certificate(".", "no_mac")
+        self.assertFalse(result)
+        result = self.cbma_adaptation._CBMAAdaptation__has_certificate(".", "fake_mac")
+        self.assertTrue(result)
+        self.patcher_has_certificate.start()
+
+    def test_wait_for_ap_no_mock(self):
+        # Not expected to have AP interface to wait by default
+        result = self.cbma_adaptation._CBMAAdaptation__wait_for_ap()
+        self.assertFalse(result)
+
+    @patch("subprocess.check_output")
+    @patch("time.sleep")
+    def test_wait_for_ap_with_mock(self, mock_sleep, mock_check_output):
+        # Fake success
+        fake_output = """
+        Interface wlan1
+                ifindex 6
+                wdev 0x1
+                addr e4:5f:01:bd:6d:cb
+                ssid comms_sleeve#6dcb
+                type AP
+                wiphy 0
+                channel 1 (2412 MHz), width: 20 MHz, center1: 2412 MHz
+                txpower 31.00 dBm
+        """
+        mock_check_output.return_value = fake_output.encode()
+        result = self.cbma_adaptation._CBMAAdaptation__wait_for_ap()
+        mock_check_output.assert_called_once_with(["iw", "dev", "wlan1", "info"])
+        self.assertTrue(result)
+        mock_sleep.assert_not_called()
+        mock_check_output.reset_mock()
+
+        def side_effect(*args, **kwargs):
+            fake_output = """
+            Interface wlan1
+                    ifindex 6
+                    wdev 0x1
+                    addr e4:5f:01:bd:6d:cb
+                    ssid comms_sleeve#6dcb
+                    type managed
```

Comment:
can't be managed with ssid :question:

Reply:
True. I'll remove ssid line from that fake output. It had just copied actual AP output and manually replaced 'AP' with 'managed'.

Reply:
fixed in commit bdd1774d11e99f5b28286a2ba5e4ebf16b9010a8